pilot/pkg/model/gateway.go

```diff
@@ -142,7 +142,7 @@ func MergeGateways(gateways []gatewayWithInstances, proxy *Proxy, ps *PushContex
 	gatewayNameForServer := make(map[*networking.Server]string)
 	verifiedCertificateReferences := sets.New()
 	http3AdvertisingRoutes := sets.New()
-	tlsHostsByPort := map[uint32]sets.Set{} // port -> host set
+	tlsHostsByPort := map[uint32]map[string]string{} // port -> host/bind map
```

Comment:
How about concat host and bind instead of double map?

Reply:
I would prefer double map which is more clear than concat host and bind.